### ISSUE TITLE
Maximum call stack size exceeded when using ngoptions

### ISSUE THREAD

**Issue by cskiwi** [krescruz/angular-materialize#49]
Hi,

I'm using the following select:

```
<select  ng-model="tune.period.value" ng-options="value.label for value in tune.period.values" material-select ></select>
```

with the following in my controller:

```
$scope.tune = [];
$scope.tune.period = {
    value: {label: "Any time", value: "any"},
    values: [
        {label: "Any time", value: "any"},
        {label: "Past 24 hours", value: "24hours"},
        {label: "Past week", value: "week"},
        {label: "Past month", value: "month"},
        {label: "Past year", value: "year"}]
};
$scope.tune.period.value = $scope.tune.period.values[0];
```

and i get this upon initialization and when seleting a value:

```
 RangeError: Maximum call stack size exceeded
    at HTMLOptionElement.n.event.dispatch (jquery-2.1.4.min.js:3)
    at HTMLOptionElement.n.event.add.r.handle (jquery-2.1.4.min.js:3)
    at Object.n.event.trigger (jquery-2.1.4.min.js:3)
    at n.fn.extend.triggerHandler (jquery-2.1.4.min.js:3)
    at Function.la.fn.on.la.cleanData (angular.js:1747)
    at n.fn.extend.remove (jquery-2.1.4.min.js:3)
    at Je.link.u.writeValue (angular.js:26172)
    at le.link.f.$render (angular.js:28124)
    at HTMLOptionElement.<anonymous> (angular.js:28243)
    at HTMLOptionElement.n.event.dispatch (jquery-2.1.4.min.js:3)
```

any clue why this is happening?

**Comment by Dlacreme** [comment]
Hi,
I'm actually experiencing the same issue with a ng-model bind to a input.
I'm using AngularJS in many projects, and I have this warning only with angular-materialize.
This kind of errors are usually generated by a $watch.

The current version is unusable, I will try an older one and wait for some help here.

**Comment by cskiwi** [comment]
I've tracked down where it came from;

when removing the following line: `$compile(element.contents())(scope);` in 

```
angular.module("ui.materialize.material_select", [])
    .directive("materialSelect", ["$compile", "$timeout", function ($compile, $timeout) {
        return {
            link: function (scope, element, attrs) {
                if (element.is("select")) {
                    $compile(element.contents())(scope);
                    $timeout(function () {
                        element.material_select();
                    });
                    if (attrs.ngModel) {
                        scope.$watch(attrs.ngModel, function () {
                            element.material_select();
                        });
                    }
                }
            }
        };
    }]);
```

the problem "disappears"

**Comment by webbiesdk** [comment]
My best bet, is that I'm bad at AngularJS. 

But anyway, it is clearly some circular-dependency, where compiling the line `$compile(element.contents())(scope);`, somehow triggers that ng-options should do something, which again triggers materialSelect (or something like that). 

You should be able to fix it, by not using ng-options (terrible fix, i know). 
Instead do like in the example in the index.html. 

But if anyone has some ideas how this can be fixed, then I'm all ears. 
(bear in mind that the materialSelect code has been slightly changed since the above comment).

**Comment by LunarDevelopment** [comment]
If you put a codepen together I'll fix that for you, I'm strapped for time at the moment though to recreate then fix..

**Comment by golyakov** [comment]
I created an example in сodepen http://codepen.io/Golyakov/pen/Vvowgo?editors=101. Please try to solve this problem.

**Comment by webbiesdk** [comment]
Fixed by #107
